File: python-puka/emit_log_direct.py
#!/usr/bin/env python
import puka
import sys

client = puka.Client("amqp://localhost/")
promise = client.connect()
client.wait(promise)


promise = client.exchange_declare(exchange='direct_logs', type='direct')
client.wait(promise)

severity = sys.argv[1] if len(sys.argv) > 1 else 'info'
message = ' '.join(sys.argv[2:]) or 'Hello World!'
promise = client.basic_publish(exchange='direct_logs', routing_key=severity,
                               body=message)
client.wait(promise)

print " [x] Sent %r:%r" % (severity, message)
client.close()

File: javascript-nodejs/src/emit_log_direct.js
#!/usr/bin/env node

const amqp = require('amqplib');

async function main() {
    const connection = await amqp.connect(process.env.AMQP_URL || 'amqp://localhost');
    const channel = await connection.createChannel();

    const exchange = 'direct_logs';
    const args = process.argv.slice(2);
    const msg = args.slice(1).join(' ') || 'Hello World!';
    const severity = (args.length > 0) ? args[0] : 'info';

    await channel.assertExchange(exchange, 'direct', {
        durable: false
    });
    channel.publish(exchange, severity, Buffer.from(msg));
    console.log(" [x] Sent %s: '%s'", severity, msg);

    setTimeout(function() {
        connection.close();
        process.exit(0);
    }, 500);
}

main();
